### PR TITLE
src: Added implementation

### PR DESCRIPTION
**Pull request by kyranet**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,9 +40,8 @@
       "ts"
     ],
     "files": [
-      "test/**/*.ts",
-      "!test/lib",
-      "!test/shared.ts"
+      "test/**/*.test.ts",
```
Comment (bdistin) [inline package.json]: please revert this change, everyone needs to be using the same thing, and that's fully linting the test folder, and the test folder acting like all the others.

Comment (kyranet) [reply package.json]: Reverted in da70fed
